analyzers/src/SonarAnalyzer.CSharp/Rules/UnusedPrivateMember.cs

```diff
@@ -463,13 +530,26 @@ private static bool IsDeclaredInPartialClass(ISymbol methodSymbol)
                 symbol.GetEffectiveAccessibility() == Accessibility.Private
                 && IsRemovable(symbol);
 
-            private static bool IsRemovableType(ISymbol typeSymbol)
+            private static bool IsRemovableType(ISymbol typeSymbol) =>
+                typeSymbol.GetEffectiveAccessibility() is var accessibility
+                && typeSymbol.ContainingType != null
+                && (accessibility == Accessibility.Private || accessibility == Accessibility.Internal)
+                && IsRemovable(typeSymbol);
+
+            private void VisitBaseTypeDeclaration(SyntaxNode node)
             {
-                var accessibility = typeSymbol.GetEffectiveAccessibility();
-                return typeSymbol.ContainingType != null
-                       && (accessibility == Accessibility.Private || accessibility == Accessibility.Internal)
-                       && IsRemovable(typeSymbol);
+                var baseTypeDeclaration = node as BaseTypeDeclarationSyntax;
+                if (ShouldCheck(baseTypeDeclaration.Modifiers, true))
+                {
+                    ConditionalStore(GetDeclaredSymbol(node), IsRemovableType);
+                }
             }
+
+            private bool ShouldCheck(SyntaxTokenList modifiers, bool checkInternal = false) =>
```

Comment:
Shouldn't this be `IsPrivate`?

Reply:
It is not only checking if it is private. I'll try to come up with a better name anyway.